config-svc/pkg/metacfg/manager.go

```diff
@@ -0,0 +1,79 @@
+package metacfg
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+type ConfigManager interface {
+	Get() Config
+	Set(config *Config) error
+}
+
+type EphemeralConfigManager struct {
+	value *Config
+	mux   sync.RWMutex
+}
+
+func (m *EphemeralConfigManager) Get() Config {
+	m.mux.RLock()
+	val := *m.value
+	m.mux.RUnlock()
+	return val
+}
+
+func (m *EphemeralConfigManager) Set(val *Config) error {
+	m.mux.Lock()
+	m.value = val
+	m.mux.Unlock()
+	return nil
+}
+
+func ReadConfigFromFile(filePath string, readOnly bool, allowDefault bool) (ConfigManager, error) {
+	if !readOnly {
+		return nil, fmt.Errorf("non-read-only config not supported yet")
```

Comment:
Hard to parse this, maybe better to say read only config are the supported option currently